### PR TITLE
Add gcm-filters to pangeo-notebook environment

### PR DESCRIPTION
**Pull request by paigem**
I'd like to add the [gcm-filters](https://gcm-filters.readthedocs.io/en/latest/index.html) package to the base notebook environment.

### CODE REVIEW

File: pangeo-notebook/environment.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ dependencies:
  - esmpy
  - fastjmd95
  - fsspec
+ - gcm-filters
```
Comment (scottyhq) [inline pangeo-notebook/environment.yml:25]: looks like this should have an underscore: `gcm_filters`

Comment (paigem) [reply pangeo-notebook/environment.yml:25]: Thanks @scottyhq. I have been successfully installing using pip on Pangeo Cloud with a dash, so figured I could use that same syntax here.

Comment (paigem) [reply pangeo-notebook/environment.yml]: Just made the update to `gcm_filters`

**Review by scottyhq — APPROVED** [review]

**Pull request merged**